vector_index.go

```diff
@@ -29,6 +29,12 @@ func init() {
 	reflectStaticSizeVectorDoc = int(reflect.TypeOf(vd).Size())
 }
 
+type VectorIndex interface {
+	Search([]float32, int64) ([]float32, []int64, error)
```

Comment:
Let's add some commentary for this, on what the expectation for all these attributes. Additionally using named parameters for the function would better the situation.